doc/running-simple-app.md

```diff
@@ -0,0 +1,198 @@
+Running Simple Applications
+===================================
+
+Introduction
+------------
+This guide explains how to run your application in Mystikos for simple development
+and testing. For a guide on how to package your application for production usage,
+refer to [doc/sign-package.md](/doc/sign-package.md). 
+
+This guide assumes you already have an application directory (appdir) populated with
+your application and any necessary dependencies. For help creating an application
+directory, refer to the user-getting-started guides (for example, 
+[doc/user-getting-started-c.md](/doc/user-getting-started-c.md)).
+
+Mounting a file system
+------------
+
+To run applications in Mystikos, we need to mount the directory containing the
+application and dependencies as a file system. There are three different
+ways to mount a file system in Mystikos.
+
+### Root file system (rootfs) mount
+
+Every application will have a root file system mount. This file system will contain 
+the entry point into your application. This guide will focus mainly on mounting a
+rootfs.
+
+
+> Unless you need to mount additional directories beyond your application directory, you will only
+need a root file system mount.
+
+### Configuration Mount
+
+This mount is specified in configuration and handled by Mystikos. File system changes do not require a new application image. Refer to [doc/design/mount-config-design.md](/doc/design/mount-config-design.md) for more details.
+
+### Explicit Mount (`mount()` syscall)
+
+Mounting can also be left to the application. Refer to
+[Mounting EXT2 images from applications](#mounting-ext2-images-from-applications-(explicit-mount-or-"app-mounted"))
+for instructions.
+
+File System Types
+------------
+
+Before we can mount the application directory, we must first generate a file system image.
+The file system image generated determines which file system type is used. There are currently
+three file systems in Mystikos. 
+
+### CPIO
+
+Choose this for small applications. The entire file system will be loaded into memory when the
+application runs. When packaging your application for production, your application directory
+will be part of the executable image.
+
+### Ext2
```

Comment:
I'd like to keep the picture in the original ext2 doc that shows the layout of the sections in ext2fs. The `file-system signature (fssig)` is the key to understand the security and trust we are trying to establish.

Reply:
Yep I verified this with the helloworld sample. I'll create a tracking item and we can revisit this.

Reply:
Added picture back to doc.